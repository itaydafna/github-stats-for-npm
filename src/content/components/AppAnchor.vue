<template>
  <div class="container">
    <div class="general">
      <IconedCounter 
        :counter-val="starsCount" 
        class="stars-count">
        <Octicon name="star"/>
      </IconedCounter>
    </div>
  </div>
</template>

<script>
import Octicon from 'vue-octicon/components/Octicon.vue';
import IconedCounter from './IconedCounter';
import 'vue-octicon/icons/star';
export default {
  components: {
    Octicon,
    IconedCounter,
  },
  props: { starsCount: Number },
  data() {
    return {};
  },
};
</script>

<style scoped>
.container {
  position: relative;
}
.general {
  position: absolute;
  display: flex;
  align-items: center;
  top: -18px;
  left: 23px;
}
.stars-count {
  margin-left: 3px;
  font-weight: 600;
}
</style>
