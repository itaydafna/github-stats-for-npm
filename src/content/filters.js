import Vue from 'vue';

export const addCommas = value => {
  if (!value) return '';
  return value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
};

export const formatDate = value => {
  try {
    if (!value) return '';
    const d = new Date(value);
    if (isNaN(d)) return '';
    return d
      .toDateString()
      .split(' ')
      .slice(1)
      .join(' ');
  } catch (e) {
    return '';
  }
};
