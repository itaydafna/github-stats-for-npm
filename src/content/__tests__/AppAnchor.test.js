import Vue from 'vue';
import { mount } from '@vue/test-utils';
import AppAnchor from '../components/AppAnchor';
import stats from './mock-stats.json';
import * as filters from '../filters';

describe('AppAnchor', () => {
  let wrapper;
  beforeEach(() => {
    Object.keys(filters).forEach(filter => {
      Vue.filter(filter, filters[filter]);
    });
    wrapper = mount(AppAnchor, { propsData: { starsCount: 4321 } });
  });
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  it('renders AppAnchor component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('adds comma separation to long numbers', () => {
    wrapper.setProps({ starsCount: 7654321 });
    expect(wrapper.html()).toContain('7,654,321');
  });

  it('does not add comma separation to short numbers', () => {
    wrapper.setProps({ starsCount: 29 });
    expect(wrapper.html()).toContain('29');
  });
});
