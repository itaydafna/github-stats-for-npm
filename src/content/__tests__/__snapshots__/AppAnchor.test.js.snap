// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppAnchor renders AppAnchor component 1`] = `
<div
  class="container"
>
  <div
    class="general"
  >
    <div
      class="stars-count"
    >
      <svg
        class="octicon"
        height="16"
        role="presentation"
        version="1.1"
        viewBox="0 0 14 16"
        width="14"
      >
        <path
          d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74z"
        />
      </svg>
       
      <span>
        4,321
      </span>
    </div>
  </div>
</div>
`;
